pages/index.js

```diff
@@ -152,12 +153,35 @@ export default function HomePage({
   }
 
   const filteredBookList = updateBookList();
-  const filterResultsCount = filteredBookList.length;
+
+  const [searchTerm, setSearchTerm] = useState("");
+  const foundBooks = filteredBookList
+    .filter((book) => {
+      const titleMatch = book.title
+        .toLowerCase()
+        .includes(searchTerm.toLowerCase());
+      const genreMatch = book.genre
+        .toLowerCase()
+        .includes(searchTerm.toLowerCase());
+      const yearMatch = book.publishYear.toString().includes(searchTerm);
+      const authorMatch = book.author
+        .toLowerCase()
+        .includes(searchTerm.toLowerCase());
+
+      return titleMatch || genreMatch || yearMatch || authorMatch;
+    })
```

Comment:
We could use object destructuring within the filter callback function for cleaner code:

```suggestion
  const foundBooks = filteredBookList
    .filter(({title, genre, publishYear, author}) => {
      const titleMatch =title
        .toLowerCase()
        .includes(searchTerm.toLowerCase());
      const genreMatch =genre
        .toLowerCase()
        .includes(searchTerm.toLowerCase());
      const yearMatch =publishYear.toString().includes(searchTerm);
      const authorMatch =author
        .toLowerCase()
        .includes(searchTerm.toLowerCase());

      return titleMatch || genreMatch || yearMatch || authorMatch;
    })
```

Reply:
Adapted